packages/cli/test/integration/role.api.test.ts

```diff
@@ -0,0 +1,86 @@
+import type { SuperAgentTest } from 'supertest';
+import * as utils from './shared/utils/';
+import { createMember } from './shared/db/users';
+import type { GlobalRole } from '@/databases/entities/User';
+import type { ProjectRole } from '@/databases/entities/ProjectRelation';
+import type { CredentialSharingRole } from '@/databases/entities/SharedCredentials';
+import type { WorkflowSharingRole } from '@/databases/entities/SharedWorkflow';
+
+const testServer = utils.setupTestServer({
+	endpointGroups: ['role'],
+});
+
+let memberAgent: SuperAgentTest;
+
+const expectedCategories = ['global', 'project', 'credential', 'workflow'] as const;
+const expectedGlobalRoles: GlobalRole[] = ['global:owner', 'global:admin', 'global:member'];
+const expectedProjectRoles: ProjectRole[] = [
+	'project:personalOwner',
+	'project:admin',
+	'project:editor',
+	'project:viewer',
+];
+const expectedCredentialRoles: CredentialSharingRole[] = ['credential:owner', 'credential:user'];
+const expectedWorkflowRoles: WorkflowSharingRole[] = [
+	'workflow:owner',
+	'workflow:editor',
+	'workflow:user',
```

Comment:
What's the difference between `editor` and `user`? Is `user` supposed to be able to click `Execute workflow` but not make changes?